evennia/contrib/wilderness.py

```diff
@@ -0,0 +1,723 @@
+        wilderness to another place inside the wilderness.
+
+        If this returns True, then the traversing can happen. Otherwise it will
+        be blocked.
+
+        This method is similar how the `at_traverse` works on normal exits.
+
+        Args:
+            traversing_object (Object): The object doing the travelling.
+            current_coordinates (tuple): (x, y) coordinates where
+                `traversing_object` currently is.
+            new_coordinates (tuple): (x, y) coordinates of where
+                `traversing_object` wants to travel to.
+
+        Returns:
+            bool: True if traversing_object is allowed to traverse
+        """
+        return True
+
+    def at_traverse(self, traversing_object, target_location):
+        """
+        This implements the actual traversal. The traverse lock has
+        already been checked (in the Exit command) at this point.
+
+        Args:
+            traversing_object (Object): Object traversing us.
+            target_location (Object): Where target is going.
+
+        Returns:
+            bool: True if the traverse is allowed to happen
+
+        """
+        itemcoordinates = self.location.wilderness.db.itemcoordinates
+
+        current_coordinates = itemcoordinates[traversing_object]
+        new_coordinates = get_new_coordinates(current_coordinates, self.key)
+
+        if not self.at_traverse_coordinates(traversing_object,
+                                            current_coordinates,
+                                            new_coordinates):
+            return False
+
+        if not traversing_object.at_before_move(None):
+            return False
+        traversing_object.location.msg_contents("{} leaves to {}".format(
+            traversing_object.key, new_coordinates),
+            exclude=[traversing_object])
+
+        self.location.wilderness.move_obj(traversing_object, new_coordinates)
+
+        traversing_object.location.msg_contents("{} arrives from {}".format(
+            traversing_object.key, current_coordinates),
+            exclude=[traversing_object])
+
+        traversing_object.at_after_move(None)
+        return True
+
+
+class WildernessMapProvider(object):
+    """
+    Default Wilderness Map provider.
+
+    This is a simple provider that just creates an infite large grid area.
+    """
+    room_typeclass = WildernessRoom
+    exit_typeclass = WildernessExit
+
+    def is_valid_coordinates(self, wilderness, coordinates):
+        """Returns True if coordinates is valid and can be walked to.
+
+        Args:
+            wilderness: the wilderness script
+            coordinates (tuple): the coordinates to check as (x, y) tuple.
+
+        Returns:
+            bool: True if the coordinates are valid
+        """
+        x, y = coordinates
+        if x < 0:
+            return False
+        if y < 0:
+            return False
+
+        return True
+
+    def get_location_name(self, coordinates):
```

Comment:
Only a get_location_name method seems very limited for the kind of use this will see. The average developer will have to at least overload the description per-room, based on the coordinate. Would it not be suitable to also supply something like `at_select(self, coordinates, caller, room)` that gets called with the room object and the character moving when the room is being prepared and allows the user to assign data to the room based on its coordinate?